packages/components/src/components/arrow-indicator/arrow-indicator.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import Icon from '../icon/icon';
+
+type TArrowIndicatorProps = {
+    classname?: string;
+    value?: number | string;
+};
+
+type TData = {
+    previous_icon: string;
+    icon: string;
+    previous_value?: string | number;
+    value?: string | number;
+};
+
+const ArrowIndicator = ({ classname, value }: TArrowIndicatorProps) => {
+    const [is_hidden, setHidden] = React.useState(true);
+    const [data, setData] = React.useState<TData>({
+        icon: '',
+        previous_icon: '',
+        value,
+    });
+    const { icon, previous_icon, previous_value } = data;
+    const has_defined_values = !isNaN(Number(previous_value)) && !isNaN(Number(value));
+    const timeout_id = React.useRef<ReturnType<typeof setTimeout>>();
+    React.useEffect(() => {
+        setHidden(false);
+
+        if (previous_value !== Number(value)) {
+            setData(prev_data => {
+                const has_increased = Number(prev_data.previous_value) < Number(value);
+                const defined_icon = has_increased ? 'IcProfit' : 'IcLoss';
+                return {
+                    icon: has_defined_values ? defined_icon : '',
+                    previous_icon: prev_data.icon,
+                    previous_value: prev_data.value,
+                    value,
+                };
+            });
```

Comment:
`usePrevious` hook didn't suit here because it memoizes a previous value which is getting overwritten after each rerender, and this component will rerender each time setState is called here and causes rerendering, which would result in the same value being saved as previous. Hence, using data from previous state instead.